test/tests/functional/pbs_node_rampdown_cset.py

```diff
@@ -0,0 +1,175 @@
+        Job.dflt_attributes[ATTR_k] = 'oe'
+
+        # skip if there are no cpuset systems in the test cluster
+        no_csetmom = True
+        for mom in self.moms.values():
+            if mom.is_cpuset_mom():
+                no_csetmom = False
+        if no_csetmom:
+            self.skipTest("Skip on cluster without cgroup cpuset system.")
+
+        # Various node names
+        self.n0 = self.moms.values()[0].shortname
+        self.n1 = self.moms.values()[1].shortname
+        self.n2 = '%s[0]' % (self.n1,)
+        self.n3 = '%s[1]' % (self.n1,)
+
+        # skip if there are less than one regular vnode and
+        # three from cpuset system (natural + 2 NUMA vnodes)
+        nodeinfo = self.server.status(NODE)
+        if len(nodeinfo) < 4:
+            self.skipTest("Not enough vnodes to run the test.")
+        # skip if second mom has less than two NUMA vnodes
+        if nodeinfo[3]['id'] != self.n3:
+            self.skipTest("Second mom has less than 2 vnodes")
+        # skip if none of the vnodes are in free state
+        for node in nodeinfo:
+            if node['state'] != 'free':
+                self.skipTest("Not all the vnodes are in free state")
+
+        self.pbs_release_nodes_cmd = os.path.join(
+            self.server.pbs_conf['PBS_EXEC'], 'bin', 'pbs_release_nodes')
+
+        # number of resource ncpus to request initially
+        ncpus = self.server.status(NODE, 'resources_available.ncpus',
+                                   id=self.n3)[0]
+        ncpus = int(ncpus['resources_available.ncpus'])
+        self.ncpus2 = ncpus / 2
+
+        # expected values upon successful job submission
+        self.job1_schedselect = \
+            "1:ncpus=1:mem=2gb+" + \
+            "1:ncpus=%d:mem=2gb:vnode=%s+" % (self.ncpus2, self.n2) + \
+            "1:ncpus=%d:mem=2gb:vnode=%s" % (self.ncpus2, self.n3)
+        self.job1_exec_host = "%s/0+%s/0*%d+%s/1*%d" % (
+            self.n0, self.n1, self.ncpus2, self.n1, self.ncpus2)
+        self.job1_exec_vnode = \
+            "(%s:ncpus=1:mem=2097152kb)+" % (self.n0,) + \
+            "(%s:ncpus=%d:mem=2097152kb)+" % (self.n2, self.ncpus2) + \
+            "(%s:ncpus=%d:mem=2097152kb)" % (self.n3, self.ncpus2)
+
+        # expected values after release of vnode
+        self.job1_schedsel1 = \
+            "1:ncpus=1:mem=2097152kb+" + \
+            "1:ncpus=%d:mem=2097152kb:vnode=%s" % (self.ncpus2, self.n2)
+        self.job1_exec_host1 = \
+            "%s/0+%s/0*%d" % (self.n0, self.n1, self.ncpus2)
+        self.job1_exec_vnode1 = \
+            "(%s:ncpus=1:mem=2097152kb)+" % (self.n0,) + \
+            "(%s:ncpus=%d:mem=2097152kb)" % (self.n2, self.ncpus2)
+
+        # cgroup cpuset path on second node
+        cmd = ['grep cgroup', '/proc/mounts', '|', 'grep cpuset', '|',
+               'grep -v', '/dev/cpuset']
+        ret = self.server.du.run_cmd(self.n1, cmd, runas=TEST_USER)
+        self.cset_path = ret['out'][0].split()[1]
```

Comment:
It seems to me that most of lines 86-118 are specifically for test_release_nodes_on_cpuset_sis(), and although I realize there is only one test in this suite right now...won't having these in the setUp make it harder to add test later?  What I'm trying to say is I think those lines would be more appropriate in the test.